### ISSUE TITLE
ic-search-bar not included in ic-dialog focus trap

### ISSUE THREAD

**Issue by GCHQ-Developer-741** [mi6/ic-ui-kit#2976]
## Summary of the bug 
When adding an `ic-search-bar` to a dialog, it cannot be tabbed to as part of the normal focus order. In fact, when it reaches that point, it stops the entire focus order entirely.

However, when searching for and selecting an option, the search bar that now has a value becomes part of the focus order again.

## 🪜 How to reproduce 
Copy code into a stackblitz or test-app:
```tsx
import { IcButton, IcDialog, IcSearchBar, IcTextField } from "@ukic/react";
import { useState } from "react";

const VanillaDialog = () => {
  const [openDialog, setOpenDialog] = useState(false);
  const handleDialogOpen = () => setOpenDialog(true);
  const handleDialogClose = () => setOpenDialog(false);

  const options = [
    { label: "Person One", value: "person one" },
    { label: "Person two", value: "person two" },
  ];

  return (
    <>
      <IcButton onClick={handleDialogOpen}>
        Launch vanilla searchbar dialog
      </IcButton>
      <IcDialog
        heading="Select a User"
        label="User selection"
        open={openDialog}
        onIcDialogClosed={handleDialogClose}
        onIcDialogConfirmed={handleDialogClose}
        onIcDialogCancelled={handleDialogClose}
      >
        <IcSearchBar
          label="What is your favourite coffee?"
          options={options}
          onIcChange={(ev) => console.log(ev.detail.value)}
          onIcSubmitSearch={(ev) => console.log(ev.detail.value)}
          placeholder="Start typing to search…"
        />
        <IcTextField label="What is your favourite coffee?" />
      </IcDialog>
    </>
  );
};

export default VanillaDialog;
```
## 📸 Screenshots or code 
https://github.com/user-attachments/assets/7b787df4-aef9-4eb4-af75-e87f757d6a86

## 🧐 Expected behaviour 
The search bar should be part of the focus order, not interrupting it, so that users can interact correctly with screen content.

## Additional info 
There is also an issue I noticed that the search icon and clear button can't be tabbed to because of the focus trap. I include this because I imagine it will be related when looked into. If not could always raise a future ticket.

**Comment by ad9242** [comment]
fix is to call the `setFocus` method on the ic-search-bar element in a similar way to how it does for ic-text-field. should be a pretty straightforward fix
